common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/azureactivedirectory/AzureActiveDirectoryAuthorizationRequest.java

```diff
@@ -22,34 +22,172 @@
+    }
+
+    public AzureActiveDirectoryPromptBehavior getPromptBehavior() {
+        return mPromptBehavior;
+    }
+
+    public void setPromptBehavior(final AzureActiveDirectoryPromptBehavior promptBehavior) {
+        mPromptBehavior = promptBehavior;
+    }
+
+    public String getClaimsChallenge() {
+        return mClaimsChallenge;
+    }
+
+    public void setClaimsChallenge(final String claimsChallenge) {
+        mClaimsChallenge = claimsChallenge;
     }
 
-    //CHECKSTYLE:OFF
     @Override
-    public String toString() {
-        return "AzureActiveDirectoryAuthorizationRequest{" +
-                "mAuthority=" + mAuthority +
-                "} " + super.toString();
+    public String getAuthorizationStartUrl() throws UnsupportedEncodingException, ClientException {
+        final String authorizationUrl = StringExtensions.appendQueryParameterToUrl(
+                getAuthorizationEndpoint(),
+                createAuthorizationRequestParameters());
+        Logger.infoPII(TAG, getCorrelationId().toString(), "Request uri to authorize endpoint is: " + authorizationUrl);
+        return authorizationUrl;
+    }
+
+    private void addDiagnosticParameters(@NonNull final Map<String, String> requestParameters) {
+        // append device and platform info in the query parameters
+        requestParameters.put(LIB_ID_PLATFORM, PLATFORM_VALUE);
+        requestParameters.put(LIB_ID_VERSION, getLibraryVersion());
+        requestParameters.put(LIB_ID_OS_VER, String.valueOf(Build.VERSION.SDK_INT));
+        requestParameters.put(LIB_ID_DM, android.os.Build.MODEL);
+    }
+
+    private void addPromptParameter(@NonNull final Map<String, String> requestParameters) {
+        // Setting prompt behavior to always will skip the cookies for webview.
+        // It is added to authorization url.
+        if (mPromptBehavior == AzureActiveDirectoryPromptBehavior.ALWAYS) {
+            requestParameters.put(QUERY_PROMPT, QUERY_PROMPT_VALUE);
+        } else if (mPromptBehavior == AzureActiveDirectoryPromptBehavior.REFRESH_SESSION) {
+            requestParameters.put(QUERY_PROMPT, QUERY_PROMPT_REFRESH_SESSION_VALUE);
+        }
+    }
+
+    private void addUserInfoParameter(@NonNull final Map<String, String> requestParameters) {
+        if (!StringExtensions.isNullOrBlank(getLoginHint())) {
+            requestParameters.put(LOGIN_HINT, getLoginHint());
+        }
+    }
+
+    private void addPkceChallengeParameters(@NonNull final Map<String, String> requestParameters) {
+        if (null != getPkceChallenge()) {
+            requestParameters.put(CODE_CHALLENGE, getPkceChallenge().getCodeChallenge());
+            // The method used to encode the code_verifier for the code_challenge parameter.
+            requestParameters.put(CODE_CHALLENGE_METHOD, getPkceChallenge().getCodeChallengeMethod());
+        }
+    }
+
+    private void addExtraQueryParameters(@NonNull final Map<String, String> requestParameters) throws ClientException {
```

Comment:
The distinction between `addExtraQueryParameters` and `appendExtraQueryParameters` is unclear -- can we solve this with naming? Javadoc?